src/blob.rs

```diff
@@ -321,10 +321,13 @@ impl<'a> BlobObject<'a> {
             }
         }
 
-        let clean = sanitize_filename_reader_friendly::sanitize(&name);
-        let mut iter = clean.splitn(2, '.');
+        let mut iter = name.splitn(2, '.');
```

Comment:
this shadows the previous `iter` definition

Reply:
i do not really understand - the `let mut iter = name.splitn(2, '.');` is the only and first `iter` definition, or?